affirm/src/main/java/com/affirm/android/model/AddressSerializer.java

```diff
@@ -0,0 +1,17 @@
+package com.affirm.android.model;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import java.lang.reflect.Type;
+
+public class AddressSerializer implements JsonSerializer<AbstractAddress> {
```

Comment:
Do we need this serializer because the AbstractAddress is abstract and not for other classes like Shipping, Billing etc?

Reply:
@kevindavmoore gson can't directly serialize abstract classes, so Address or CAAddress will be ignored when call `gson.toJson (checkout)`, because they both extends from AbstractAddress. 
(AbstractAddress is added for support Canada, i added CAAddress and extends from AbstractAddress)